test/format/format.sh

```diff
@@ -0,0 +1,301 @@
+		[[ "$jobs" =~ ^[1-9][0-9]*$ ]] || {
+			echo "$0: -j option argument must be a non-zero integer"
+			exit 1
+		}
+		shift ; shift ;;
+	-n)
+		runs="$2"
+		[[ "$runs" =~ ^[1-9][0-9]*$ ]] || {
+			echo "$0: -n option argument must be an non-zero integer"
+			exit 1
+		}
+		shift ; shift ;;
+	-S)
+		smoke_test=1
+		shift ;;
+	-t)
+		minutes="$2"
+		[[ "$minutes" =~ ^[1-9][0-9]*$ ]] || {
+			echo "$0: -t option argument must be a non-zero integer"
+			exit 1
+		}
+		shift ; shift ;;
+	-v)
+		verbose=1
+		shift ;;
+	-*)
+		usage ;;
+	*)
+		break ;;
+	esac
+done
+format_args="$*"
+
+# Find the format and wt binaries (the latter is only required for abort/recovery testing).
+# Find the configuration file and run directory.
+build=$(dirname $0)
+format_binary="$build/t"
+wt_binary="$build/../../wt"
+[[ ! -x "$format_binary" ]] && {
+	echo "$0: format program \"$format_binary\" not found"
+	exit 1
+}
+[[ $abort_test -ne 0 ]] && [[ ! -x "$wt_binary" ]] && {
+	echo "$0: wt program \"$wt_binary\" not found"
+	exit 1
+}
+[[ -z "$config" ]] && config="$build/CONFIG.stress"
+[[ -f "$config" ]] || {
+	echo "$0: configuration file \"$config\" not found"
+	exit 1
+}
+[[ -d "$home" ]] || {
+	echo "$0: run directory \"$home\" not found"
+	exit 1
+}
+
+[[ $verbose -ne 0 ]] &&
+    echo "$0 [-c $config] [-h $home] [-j $jobs] [-n $runs] [-t $minutes] $format_args"
+
+failure=0
+success=0
+# Resolve/cleanup completed jobs.
+resolve()
+{
+	for dir in $home/RUNDIR.[0-9]*; do
+		# Skip log files and failures we've already reported.
+		[[ ! -d $dir ]] || [[ -f "$dir/reported" ]] && continue
+		log="$dir.log"
+
+		# Discard successful runs.
+		grep 'successful run completed' $log > /dev/null && {
+			rm -rf $dir $log
+			success=$(($success + 1))
+			[[ $verbose -ne 0 ]] && echo "$0: job in $dir successfully completed"
+			continue
+		}
+
+		# Test recovery on runs configured for random abort. */
+		grep 'aborting to test recovery' $log > /dev/null && {
+			cp -pr $dir $dir.RECOVER
+
+			(echo
+			 echo "$0: running recovery after abort test"
+			 echo "$0: original directory copied into $dir.RECOVER"
+			 echo) >> $log
+
+			# Everything is a table unless explicitly a file.
+			uri="table:wt"
+			grep 'data_source=file' $dir/CONFIG > /dev/null && uri="file:wt"
+						
+			# Use the wt utility to recover & verify the object.
+			if  $($wt_binary -R -h $dir verify $uri >> $log 2>&1); then
+				rm -rf $dir $dir.RECOVER $log
+				success=$(($success + 1))
+				[[ $verbose -ne 0 ]] &&
+				    echo "$0: job in $dir successfully completed"
+			else
+				echo "$0: failure status reported" > $dir/reported
+				failure=$(($failure + 1))
+				echo "$0: job in $dir failed abort/recovery testing"
+			fi
+			continue
+		}
+
+		# Discard runs where the timer went off.
+		grep 'caught signal' $log > /dev/null && {
+			rm -rf $dir $log
+			[[ $verbose -ne 0 ]] && echo "$0: job in $dir aborted"
+			continue
+		}
+
+		# Report failures.
+		grep 'RUN PARAMETERS' $log > /dev/null && {
```

Comment:
It's not super clear to me why `'RUN PARAMETERS'` indicates a failure. Some further comment would be helpful.

Reply:
I changed `format` to output an explicit failure message instead of checking for `RUN PARAMETERS`, fixed in 5dff87605.

Reply:
Looks better, thanks.